csrc/attention/attention_kernels.cu

```diff
@@ -116,12 +117,15 @@ __global__ void single_query_cached_kv_attention_kernel(
   // th vectors of the query, and so on.
   // NOTE(woosuk): Because q is split from a qkv tensor, it may not be contiguous.
   const scalar_t* q_ptr = q + seq_idx * q_stride + head_idx * HEAD_SIZE;
-  Q_vec q_vecs[NUM_VECS_PER_THREAD];
+  __shared__ Q_vec q_vecs[THREAD_GROUP_SIZE][NUM_VECS_PER_THREAD];
+  if (thread_group_idx <= NUM_THREAD_GROUPS_LOWER_BOUND) {
```

Comment:
True, it's there to ensure a bug doesn't enter in case somehow this changes in the future. Want me to remove it? I am betting the compiler removes it anyhow, so it doesn't really affect the code that actually runs.